logging/net_appender.go

```diff
@@ -0,0 +1,324 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+	apppb "go.viam.com/api/app/v1"
+	"go.viam.com/utils"
+	"go.viam.com/utils/protoutils"
+	"go.viam.com/utils/rpc"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	defaultMaxQueueSize = 20000
+	writeBatchSize      = 100
+)
+
+// CloudConfig contains the necessary inputs to send logs to the app backend over grpc.
+type CloudConfig struct {
```

Comment:
The old net_logger, (inside the web/server package) directly used the `config` package's `Config.Cloud` [structure](https://github.com/viamrobotics/rdk/blob/c5bfc3ef574220cbea0d0e8c973908d4b908e8a8/config/config.go#L31). But with this code living inside the logging package, this creates a cycle. Logging depends on config and config depends on logging.

The config package is a big ball of low-level data structures along with a bunch of business logic about parsing configs and making connections, and having those connections output errors to loggers.

I had to sever some cycles while avoiding splitting out the `config` package into separate data and code packages.